assets/hole.js

```diff
@@ -58,7 +58,11 @@ onload = () => {
         if (!langs.find(l => l.id == lang))
             location.hash = lang = 'python';
 
-        cm.setOption('mode', {name: lang == 'c' ? 'clike' : lang, startOpen: true});
+        if (lang == 'c' || lang == 'java')
+            cm.setOption('mode', {name: 'text/x-' + lang, startOpen: true});
+        else
+            cm.setOption('mode', {name: lang, startOpen: true});
+
```

Comment:
If you want, in a separate changelist we can clean this up by switching to using mime types for all of the languages.